game_board.gd

```diff
@@ -308,26 +329,30 @@ func _on_cursor_accept_pressed(cell: Vector2) -> void:
 	# The cursor's "accept_pressed" means that the player wants to interact with a cell. Depending
 	# on the board's current state, this interaction means either that we want to select a unit all
 	# that we want to give it a move order.
-	print("active unit", _active_unit)
-	if not _active_unit:
-		_select_active_unit(cell)
-	elif _active_unit.isSelected:
-		print("is_occupied(cell) = ", is_occupied(cell))
-		if is_occupied(cell) && _units[cell] != _active_unit:
-			print('selecting target', _units[cell])
-			_select_target_unit(cell)
-			return
-		if _active_unit.cell != cell:
-			print("_active_unit.cell", _active_unit.cell)
-			print("cell", cell)
-			print("CELL IS NOT ACTIVE UNIT")
-			_move_active_unit(cell)
+	print("turn taker: ", turnTracker.currentTurnTaker)
+	if turnTracker.currentTurnTaker == Global.turnTakers.PLAYER:
+		print("active unit: ", _active_unit)
+		if not _active_unit:
+			_select_active_unit(cell)
+		elif _active_unit.isSelected and _active_unit.actionPoints > 0:
+			print("elif ap= ", _active_unit.actionPoints)
+			if is_occupied(cell) && _units[cell] != _active_unit:
+				print("selecting...")
+				_select_target_unit(cell)
+				return
+			if _active_unit.cell != cell:
+				print("move")
+				_move_active_unit(cell)
 
 func _unhandled_input(event: InputEvent) -> void:
-	if _active_unit and event.is_action_pressed("ui_cancel"):
+	if _active_unit and event.is_action_pressed("ui_cancel") && _active_unit.actionPoints == 3:
 		_deselect_active_unit()
 		_clear_active_unit()
 
 
 func _on_unit_base_updated():
 	_reinitialize()
+
+
+func _on_cursor_skip_turn():
+	turnTracker._on_unit_turn_finished()
```

Comment:
very useful debugging tool, should be removed eventually or at least retooled into a menu option